test/consistent-package-json.test.ts

```diff
@@ -0,0 +1,142 @@
+import {dirname, join, relative, resolve} from 'path';
+
+import {readJSONSync} from 'fs-extra';
+import glob from 'glob';
+
+const KNOWN_TEMPLATE_KEYS = [
+  'author',
+  'bugs',
+  'dependencies',
+  'description',
+  'files',
+  'homepage',
+  'license',
+  'main',
+  'name',
+  'publishConfig',
+  'repository',
+  'scripts',
+  'sideEffects',
+  'types',
+  'version',
+];
+
+const ROOT_PATH = resolve(__dirname, '..');
+const packageJSONTemplatePath = join('templates', 'package.hbs.json');
+const packageJSONTemplate = readJSONSync(
+  join(ROOT_PATH, packageJSONTemplatePath),
+);
+
+// eslint-disable-next-line jest/valid-describe
+describe(packageJSONTemplatePath, () => {
+  it('has only known keys', () => {
+    const keys = Object.keys(packageJSONTemplate).sort();
+
+    // Template keys should match exactly. If updating it, remember to update tests!
+    expect(keys).toStrictEqual(KNOWN_TEMPLATE_KEYS);
+  });
+});
+
+readPackages().forEach(({packageName, packageJSON, packageJSONPath}) => {
+  // eslint-disable-next-line jest/valid-describe
+  describe(packageJSONPath, () => {
+    it('specifies Shopify as author', () => {
+      expect(packageJSON.author).toBe(packageJSONTemplate.author);
+    });
+
+    it('specifies Quilt Issues as bugs URL', () => {
+      expect(packageJSON.bugs).toStrictEqual(packageJSONTemplate.bugs);
+    });
+
+    it('specifies a dependencies', () => {
+      expect(packageJSON.dependencies).not.toStrictEqual({});
+    });
+
+    it('specifies a description', () => {
+      expect(packageJSON.description).not.toBeUndefined();
+    });
+
+    it('specifies files', () => {
+      if(packageName == 'polyfills') return; // FIXME
+
+      expect(packageJSON.files).toContain('dist/*');
+      expect(packageJSON.files).toContain('!tsconfig.tsbuildinfo');
+    });
+
+    it('specifies Quilt deep-link homepage', () => {
+      const expectedHomepage = compile(packageJSONTemplate.homepage);
+
+      expect(packageJSON.homepage).toBe(expectedHomepage);
+    });
+
+    it('specifies the MIT license', () => {
+      expect(packageJSON.license).toBe(packageJSONTemplate.license);
```

Comment:
```suggestion
      expect(packageJSON.license).toBe('MIT');
```

Reply:
I had originally hardcoded things like this in this file, but then I started adding tests for the template as well. I realized that it made more sense to treat the template as the expected values, and ensure each package conforms to it.

If we hard code `MIT` here, it means that if we change the template down the road, the change will only be detected by the test suite after generating a new package and seeing it fail the suite. By coupling to the template, changing the license there would cause every package's license test to fail, which is likely what we want, since we'd probably want to update their licenses as well.